src/freenas/usr/local/lib/middlewared_truenas/plugins/enclosure_/r50_nvme.py

```diff
@@ -18,34 +15,41 @@ class EnclosureService(Service):
     @private
     def r50_nvme_enclosures(self):
         system_product = self.middleware.call_sync("system.info")["system_product"]
-        if system_product != "TRUENAS-R50":
+        if system_product not in ("TRUENAS-R50", "TRUENAS-R50B"):
             return []
-
-        nvme_to_nvd = self.middleware.call_sync('disk.nvme_to_nvd_map')
+        else:
+            model = system_product.split("-")[-1].upper()
+            if model == "R50B":
+                self.HANDLES[r"\_SB_.PC00.RP01.PXSX"] = 1
 
         slot_to_nvd = {}
-        for nvme, nvd in nvme_to_nvd.items():
+        for nvme, nvd in self.middleware.call_sync('disk.nvme_to_nvd_map').items():
             try:
                 location = sysctl.filter(f"dev.nvme.{nvme}.%location")[0].value
                 m = re.search(self.RE_HANDLE, location)
-                if not m:
-                    continue
-
-                handle = m.group(1)
```

Comment:
This should be slot 0 and slot 2 should be the empty one. I believe based upon output